### PR TITLE
stm32h7x3: ADC

### PR DESCRIPTION
**Pull request by richardeoin**
Definitions and fixes for ADC

For H742/743/750/753 Revision Y ('current' silicon), although Revision Y
specifics are broken out into a separate file.

Written against RM0433 Rev 6 04-Apr-2019

### CODE REVIEW

File: peripherals/adc/adc_v3.yaml [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+      Cycles2_5: [1, "2.5 ADC clock cycles"]
+      Cycles8_5: [2, "8.5 ADC clock cycles"]
+      Cycles16_5: [3, "16.5 ADC clock cycles"]
+      Cycles32_5: [4, "32.5 ADC clock cycles"]
+      Cycles64_5: [5, "64.5 ADC clock cycles"]
+      Cycles387_5: [6, "387.5 ADC clock cycles"]
+      Cycles810_5: [7, "810.5 ADC clock cycles"]
+  # Channel preselection
+  PCSEL:
+    "PCSEL*":
```
Comment (adamgreig) [inline peripherals/adc/adc_v3.yaml]: All the `PCSEL` bits are plain read-write so I think should have an enum for 0 (to un-preselect a channel) and not be in `_write`?

File: peripherals/adc/adc_h7_revision_y.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    RES:
+      SixteenBit: [0, "16-bit resolution"]
+      FourteenBit: [1, "14-bit resolution"]
+      TwelveBit: [2, "12-bit resolution"]
+      TenBit: [3, "10-bit resolution"]
+      EightBit: [4, "8-bit resolution"]
+  CR:
+    BOOST:
+      "On": [0, "Boost mode off. Used when ADC clock < 20MHz"]
+      "Off": [1, "Boost mode on. Used when ADC clock > 20MHz"]
```
[the remark this answers is not in the document]
Comment (richardeoin) [reply peripherals/adc/adc_h7_revision_y.yaml]: I agree! Also it'll need completely different variant names for Rev V

File: peripherals/adc/adc_v3.yaml [hunk 3]
```diff
@@ -0,0 +1,246 @@
+# ADC as used on H7
+
+"ADC?":
+  ISR:
```
[the remark this answers is not in the document]
Comment (richardeoin) [reply peripherals/adc/adc_v3.yaml]: Okay, yes that's a more accurate capture of the RM and likely usage

File: peripherals/adc/adc_v3.yaml (hunk 1 above)
Comment (richardeoin) [reply peripherals/adc/adc_v3.yaml]: I was making an assumption that the only valid usage is to preselect... However this isn't the place to restrict something that isn't restricted in the RM. Not even sure if my assumption is valid either!

**Pull request merged**